tools/c7n_oci/c7n_oci/provider.py

```diff
@@ -21,13 +47,68 @@ class OCI(Provider):
     resource_map = ResourceMap
 
     def initialize(self, options):
+        if options["account_id"] is None:
+            options["oci_config"] = get_oci_profile_session(options).get_config()
         return options
 
+    def _get_subscribed_region(self, options):
+        client = get_oci_profile_session(options).client("oci.identity.IdentityClient")
+        return client.list_region_subscriptions(tenancy_id=options["oci_config"]["tenancy"]).data
+
+    def _is_subscribed_region(self, subscribed_regions, region, policy_collection):
+        for r in subscribed_regions:
+            if r.region_name == region and r.status == "READY":
+                return True
+
+        policy_collection.log.info(
+            (
+                "User's tenancy is not subscribed to the %s region. "
+                "So skipping this region for the policy execution."
+            ),
+            region,
+        )
+        return False
+
     def initialize_policies(self, policy_collection, options):
+        from c7n.policy import Policy, PolicyCollection
+
+        if options.regions:
```

Comment:
this block could be dedented by handling the other case in the conditional and returning policy_collection.

Reply:
Done